File: leetcode2/1easy/이총명/Q1909.java
class Solution {
    public boolean canBeIncreasing(int[] nums) {
        int count = 0;

        for(int i = 1; i < nums.length; i++){
            if(nums[i] <= nums[i-1]) {
                count++;
                if(count > 1) return false;
            }

            if(i - 2 >= 0 && nums[i] <= nums[i-2]) {
                nums[i] = nums[i - 1];
            }
        }
        return true;
    }
}

File: leetcode2/1easy/강병호/Q1909.py
class Solution:
    def canBeIncreasing(self, nums: list[int]) -> bool:
        
        remove_cnt = 1
        i = 1
        n = len(nums)

        while True:
            print(f"i : {i}")
            if i == 0:
                i+=1
            if i == len(nums) - 1:
                break
            if nums[i-1] >= nums[i] and not remove_cnt:
                return False
            elif nums[i-1] >= nums[i] and remove_cnt:
                nums.pop(i-1)
                i = i - 1
                remove_cnt -= 1
                continue

            i += 1

        return True
